staging/kos/pkg/apis/network/types.go

```diff
@@ -30,6 +31,32 @@ type ExtendedObjectMeta struct {
 	// +listMapKey=section
 	// +optional
 	Writes WriteSet
+
+	// LastClientWrite identifies the latest client write among those needed to
+	// implement the API object. Unlike `Writes`, it might describe a write on
+	// an API object other than the owning API object.
+	// +optional
+	LastClientWrite ClientWrite
+
+	// LastControllerStart identifies the relevant controller that started last.
+	// Relevant controllers are those directly or indirectly responsible for
+	// implementing the API object.
+	// +optional
+	LastControllerStart ControllerStart
+}
+
+// Validate returns a list of errors carrying information on why
+// the ExtendedObjectMeta is invalid, or an empty list if the ExtendedObjectMeta
+// is valid.
+func (eom ExtendedObjectMeta) Validate() field.ErrorList {
```

Comment:
In the same package makes for the same problem, as far as much tooling is concerned, since golang is very much more concerned with packages than files.  In Kubernetes the helpers are published in a different repo under k8s.io.  Here we do not have multiple repos, so a sufficiently distinct package seems like the right answer to me.

Reply:
Done in 9e5ad76 . There are also other helper functions that are unrelated to this PR and live in the same pkg as type definitions; those are not moved to keep this PR focused. After this PR I plan one housekeeping PR to do some refactorings and make more concise some parts of the code; the factor out we're discussing will be done in that occasion.